### ISSUE TITLE
Untracked working tree files blocking updates

### ISSUE THREAD

**Issue by pornel** [composer/composer#2896]
Composer, when told to throw away changes blocking an update, fails to checkout the update if there is any file or local change preventing git from doing so.

```
[RuntimeException]
  Failed to execute git checkout 'xxx' && git reset --hard 'xxx'

  error: The following untracked working tree files would be overwritten by checkout:
  	some_rubbish_temp_file
  Please move or remove them before you can switch branches.
  Aborting
```

This can be fixed in two ways:

* when stashing, add `--include-untracked` flag which will stash newly added files as well.
* when checking out, add `--force` flag which will overwrite files instead of aborting.

**Comment by glennmschultz** [comment]
god damit stuck by this thing.

**Comment by mcd-php** [comment]
@pornel , @CaptainQuirk I'm not from composer team and so can't talk in their name, but I believe what dirs/files not created by me and not tracked by VCS, be it dependencies, auto-generated code or documentation, should be capable to be completely purged by build automation scripts without any harm for the project. Nothing what you want to keep should be in that folders/files.

**Comment by Seldaek** [comment]
Fixed by #3789
